### PR TITLE
[FLINK-29655] Split Flink CRD from flink-kubernates-operator module

### PR DESCRIPTION
**Pull request by mbalassi**
This refactor introduces a new module called `flink-kubernetes-operator-api` and moves the CRD in said module with the aim of making the transitive dependencies of it as slim as possible via removing flink from the compile time dependency tree.

### CODE REVIEW

File: flink-kubernetes-operator-api/src/main/java/org/apache/flink/kubernetes/operator/crd/status/ReconciliationMetadata.java [hunk 1]
```diff
@@ -15,10 +15,9 @@
  * limitations under the License.
  */
 
-package org.apache.flink.kubernetes.operator.reconciler;
+package org.apache.flink.kubernetes.operator.crd.status;
```
Comment (gyfora) [inline flink-kubernetes-operator-api/src/main/java/org/apache/flink/kubernetes/operator/crd/status/ReconciliationMetadata.java:18]: I suggest keeping this out of the `crd` package. It is internal to the operator logic and moving it under the CRD pollutes the CRD ref docs

File: flink-kubernetes-operator/pom.xml [hunk 2]
```diff
@@ -195,32 +202,6 @@ under the License.
         </resources>
 
         <plugins>
-            <plugin>
```
Comment (gyfora) [inline flink-kubernetes-operator/pom.xml:198]: We should also move the `maven-antrun-plugin` usages for the backward compatibility check to the api plugin

File: flink-kubernetes-operator-api/src/main/java/org/apache/flink/kubernetes/operator/crd/status/ReconciliationMetadata.java (hunk 1 above)
Comment (mbalassi) [reply flink-kubernetes-operator-api/src/main/java/org/apache/flink/kubernetes/operator/crd/status/ReconciliationMetadata.java]: Good catch, done.

File: flink-kubernetes-operator/pom.xml (hunk 2 above)
Comment (mbalassi) [reply flink-kubernetes-operator/pom.xml:198]: Done.

**Comment by mbalassi** [comment]
@tweise has pointed out to me that we actually do not guarantee Java API compatibility at this point, and consequently it is best to make the changes now necessary to be able to provide said guarantee later.
https://nightlies.apache.org/flink/flink-kubernetes-operator-docs-main/docs/operations/compatibility/#java-api-compatibility

**Review by gyfora — COMMENTED** [review]
Having both `api` and `crd` in the package name feels a bit redundant for me.

CRD does not really mean anything in the java world, and means api in the kubernetes world. I   don't feel strongly about renaming the packages but if we are going to do it I think we should simply change crd -> api

**Comment by mbalassi** [comment]
Thanks, @gyfora. Moved the packages as you suggested.

**Pull request merged**